### PR TITLE
Add tests for configurations

### LINKED ISSUE #1891: Add test coverage for configurations
There is no test coverage for configurations

### PR DESCRIPTION
**Pull request by diversemix**
#### Background
Add tests for configurations

#### Any relevant tickets

Closes #1891

#### How has this been tested?

Tests added

### CODE REVIEW

File: config/__tests__/__snapshots__/config.prod.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,519 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`prod configuration should load correct configuration: prod 1`] = `
+Config {
+  "authsome": Object {
+    "isPublic": true,
+    "mode": "/home/peter/dev/elife-xpub/config/non-serializable/authsome.js",
```
Comment (giorgiosironi) [inline config/__tests__/__snapshots__/config.prod.test.js.snap]: besides helping to commit credentials, the snapshots seems also dependent on the machine where they were generated; so as for the other PR, I'd remove them and only check correct loading, not values

Comment (diversemix) [reply config/__tests__/__snapshots__/config.prod.test.js.snap]: @giorgiosironi ... check it out now

**Review by will-byrne — APPROVED** [review]

File: config/__tests__/testConfig.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+module.exports = configName => {
+  process.env.NODE_CONFIG_ENV = configName
+  const configObject = require('config')
+  let config = _.cloneDeep(configObject)
+  expect(config.configTag).toBe(configName)
+
+  config = getKeys(config)
+  config = config.filter(value => value !== 'pubsweet-server.secret')
```
Comment (giorgiosironi) [inline config/__tests__/testConfig.js]: why this secret is excluded and the rest like the AWS credentials are listed in the snapshots?

File: config/__tests__/__snapshots__/config.test.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  "pubsweet.components.4",
+  "pubsweet.components.5",
+  "pubsweet.components.6",
+  "pubsweet.components.7",
+  "pubsweet.components.8",
+  "pubsweet.components.9",
+  "pubsweet.components.10",
+  "pubsweet.components.11",
+  "pubsweet.components.12",
+  "pubsweet.components.13",
```
Comment (giorgiosironi) [inline config/__tests__/__snapshots__/config.test.test.js.snap]: if you really want to maintain these lists everytime a configuration key changes, up to you

Comment (diversemix) [reply config/__tests__/__snapshots__/config.test.test.js.snap]: We can alter this test as we go ... at least we have one now :)

File: config/__tests__/testConfig.js (hunk 2 above)
Comment (diversemix) [reply config/__tests__/testConfig.js]: done ... please check!

File: config/__tests__/__snapshots__/config.test.test.js.snap (hunk 3 above)
Comment (diversemix) [reply config/__tests__/__snapshots__/config.test.test.js.snap]: done - please check!

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**